src/gui/editors/AutomationEditor.cpp

```diff
@@ -488,130 +494,180 @@ void AutomationEditor::mousePressEvent( QMouseEvent* mouseEvent )
-					MidiTime valuePos(posTicks);
+					// Update the last clicked position so we remove all nodes from
+					// that point up to the point we release the mouse button
+					m_drawLastTick = posTicks;
 
-					// Starts actually moving/draging the node
-					MidiTime newTime = m_pattern->setDragValue(valuePos,
-						level, true, mouseEvent->modifiers() & Qt::ControlModifier);
+					// If we right-clicked a node, remove it
+					if (clickedNode != tm.end())
+					{
+						m_pattern->removeValue(clickedNode.key());
+						Engine::getSong()->setModified();
+					}
 
-					// Set the iterator to our newly created node so we can use it later
-					// for the m_moveXOffset calculation
-					clickedNode = tm.find(newTime);
+					m_action = ERASE_VALUES;
 				}
-				else // If we clicked over an existing node
+				else if(m_mouseDownRight) // And right click resets outValues
 				{
-					// Simply start moving/draging it
-					MidiTime newTime = m_pattern->setDragValue(MidiTime(clickedNode.key()),
-						level, true, mouseEvent->modifiers() & Qt::ControlModifier);
+					// So we actually don't care if we clicked the node itself, but if we clicked
+					// the node representing its outValue
+					clickedNode = getNodeAt(mouseEvent->x(), mouseEvent->y(), true);
 
-					// We need to update our iterator because setDragValue removes the node that
-					// is being dragged, so if we don't update it we have a bogus iterator
-					clickedNode = tm.find(newTime);
-				}
+					// If we clicked an outValue reset it
+					if (clickedNode != tm.end())
+					{
+						clickedNode.value().setOutValue(
+							clickedNode.value().getInValue());
+						Engine::getSong()->setModified();
+					}
 
-				// Set the action to MOVE_VALUE so moveMouseEvent() knows we are moving a node
-				m_action = MOVE_VALUE;
+					// Update the last clicked position so we reset all outValues from
+					// that point up to the point we release the mouse button
+					m_drawLastTick = posTicks;
 
-				// Calculate the offset from the place the mouse click happened in comparison
-				// to the center of the node
-				int alignedX = (int) ((float) ((clickedNode.key() - m_currentPosition) * m_ppb) /
-							MidiTime::ticksPerBar());
-				m_moveXOffset = x - alignedX - 1;
+					m_action = RESET_OUTVALUES;
+				}
+			break;
+			}
+			case DRAW_OUTVALUES:
+			{
+				m_pattern->addJournalCheckPoint();
 
-				// Set move-cursor
-				QCursor c(Qt::SizeAllCursor);
-				QApplication::setOverrideCursor(c);
+				// On this mode, left click sets the outValue
+				if (m_mouseDownLeft)
+				{
+					// If we are pressing alt, we want to drag the outValue of a node
 
-				// Update the last clicked position so it can be used if we draw a line later
-				m_drawLastTick = posTicks;
-				m_drawLastLevel = level;
-			}
+					// So we actually don't care if we clicked the node itself, but if we clicked
+					// the node representing its outValue
+					clickedNode = getNodeAt(mouseEvent->x(), mouseEvent->y(), true);
```

Comment:
We can pull this a level up since it's used in both branches of the if statement. This might help with merging lower if statements as well. We're getting pretty deep in levels here.

Reply:
I moved it a level up. There's a possibility of merging one of the conditionals (`if (m_mouseDownLeft && clickedNode != tm.end())` and similarly below), but personally I think this would cost on how understandable the code is going to be. I find it more organized to have a block for each mouse button and then blocks for the particular cases to check for those buttons. But hit me up if you think we should merge those conditionals anyways!